operator/cilium_node.go

```diff
@@ -409,12 +424,28 @@ func runCNPNodeStatusGC(name string, clusterwide bool, nodeStore cache.Store) {
 
 }
 
-func updateCNP(ciliumClient v2.CiliumV2Interface, cnp *cilium_v2.CiliumNetworkPolicy, nodesToDelete map[string]v1.Time) {
-	if len(nodesToDelete) == 0 {
+func clearStatusNodesInCnp(ciliumClient v2.CiliumV2Interface, cnp *cilium_v2.CiliumNetworkPolicy) {
+	data, err := json.Marshal([]k8s.JSONPatch{
+		{
+			OP:   "remove",
+			Path: "/status/nodes",
+		},
+	})
+	if err != nil {
+		log.WithError(err).Debug("Unable to json marshal patch object")
```

Comment:
I think it's a bit clunky to define this here. Rather, having `patchStatusInCNP()` take a `[]k8s.JSONPatch` object seems cleaner to me. That way the marshalling can be done in `patchStatusInCNP()` and if there's an error, then it can be returned.